### PR TITLE
Chores: upload credential-provider-config.yaml to artifact store in ci build

### PR DESCRIPTION
**Pull request by MartinForReal**
**What type of PR is this?**
/kind failing-test
/kind cleanup

**What this PR does / why we need it**:

Uploading credentials-config yaml to azure blob storage and fix flaky e2e scenarios which uses oot azure credential provider.  

In recent e2e build of cloud provider project, Cluster node may fail to initialize the control plane because credential provider config is invalid and kubelet crashes.

```
Nov 15 12:40:06.473151 capz-oh7kxt-control-plane-klvq7 kubelet[2278]: E1115 12:40:06.473135    2278 kuberuntime_manager.go:274] "Failed to register CRI auth plugins" err="error decoding config /var/lib/kubelet/credential-provider-config.yaml: Object 'Kind' is missing in '404: Not Found'"
```

The config is downloaded from git repo when the node is provisioned. 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/c6c977e6d3b4407136d0abcb1a30da10451090ac/templates/test/ci/cluster-template-prow-ci-version.yaml#L116

But the commit may reside in forked repo. And provision script will get 404 when it tries to download config file in this scenario.
The config file may be a part of changes. We should upload it to blob store to make sure config file is validated too.

Also, status code of http response is sent to stdout for debug purpose. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by sonasingh46 — COMMENTED** [review]
/lgtm

File: templates/test/ci/prow-ci-version/patches/oot-credential-provider-win.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@
       mkdir C:\var\lib\kubelet\credential-provider
       curl.exe --retry 10 --retry-delay 5 -L "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${IMAGE_TAG_ACR_CREDENTIAL_PROVIDER}/azure-acr-credential-provider.exe" --output C:\var\lib\kubelet\credential-provider\acr-credential-provider.exe
       cp C:\var\lib\kubelet\credential-provider\acr-credential-provider.exe C:\var\lib\kubelet\credential-provider\acr-credential-provider
-      curl.exe --retry 10 --retry-delay 5 -L https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/${IMAGE_TAG_ACR_CREDENTIAL_PROVIDER}/examples/out-of-tree/credential-provider-config-win.yaml --output C:\var\lib\kubelet\credential-provider-config.yaml
+      curl.exe --retry 10 --retry-delay 5 -L "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${IMAGE_TAG_ACR_CREDENTIAL_PROVIDER}/credential-provider-config-win.yaml" --output C:\var\lib\kubelet\credential-provider-config.yaml
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply templates/test/ci/prow-ci-version/patches/oot-credential-provider-win.yaml:11]: no for now..

[the remark this answers is not in the document]
Comment (sonasingh46) [reply templates/test/ci/prow-ci-version/patches/oot-credential-provider-win.yaml:11]: sure.

**Comment by sonasingh46** [comment]
@MartinForReal -- Can you please write a PR description for record, something on the lines of why this change is needed so that folks can get some context who may be seeing for the first time? 

Something on lines -- uploading credentials-config yaml to azure blob storage and using that in CI from azure blob storage and not from the github repository so that the downloads are fast and not flaky and hence stable CI ??

**Comment by sonasingh46** [comment]
Also Netlify deploy is failing? I did not get an option to re trigger this. Does anyone know the root issue or how to re trigger this? cc @mboersma

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve
/cherry-pick release-1.11

**Pull request merged**